include/deal.II/lac/vector.templates.h

```diff
@@ -48,6 +48,20 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+/**
+ * Helper struct to tell us if we can use SIMD instructions for the given @p
+ * Number type.
+ */
+template <typename Number>
+struct UseSimdFor
+{
+    static const bool value;
+};
+
+template <typename Number>
+const bool UseSimdFor<Number>::value = true;
```

Comment:
For partial specializations, we typically define the 'true' value directly in the struct and create a specialization class template <>
struct UseSimdFor<long double> {...}